### PR TITLE
MINOR: Documentation improvements

### PR DESCRIPTION
**Pull request by ijuma**
- Fix typo in api.html
- Mark security features as beta quality (similar to new consumer). Is there better wording?
- Improve wording and clarify things in a number of places
- Improve layout of `pre` blocks (tested locally, which doesn't seem to use the same stylesheets as the deployed version)
- Use producer.config in console-producer.sh command
- Improve SASL documentation structure

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @junrao, @gwenshap and @edenhill

**Comment by ijuma** [comment]
Test failure is unrelated as I didn't change any code in this PR.

File: docs/security.html [hunk 1]
```diff
@@ -406,10 +419,10 @@
 	<li> Set the configuration property <tt>zookeeper.set.acl</tt> in each broker to true</li>
 </ol>
 
-The metadata stored in ZooKeeper is such that only brokers will be able to modify the corresponding znodes, but znodes are world readable. The rationale behind this decision is that the data stored in ZooKeeper is not sensitive, but inappropriate manipulation of znodes can cause cluster disruption.
+The metadata stored in ZooKeeper is such that only brokers will be able to modify the corresponding znodes, but znodes are world readable. The rationale behind this decision is that the data stored in ZooKeeper is not sensitive, but inappropriate manipulation of znodes can cause cluster disruption. We also recommend limiting the access to ZooKeeper via network segmentation (only the brokers need access to ZooKeeper if the new consumer and new producer are used).
```
Comment (junrao) [inline docs/security.html]: Just a minor clarification: "only brokers and some admin tools need access to zookeeper".

**Comment by junrao** [comment]
Thanks for the patch. Just a minor comment above. Otherwise, LGTM.

File: docs/security.html (hunk 1 above)
Comment (ijuma) [inline docs/security.html]: Good point @junrao, fixed.

**Comment by ijuma** [comment]
@edenhill, can you please check the latest version? I also changed the client example as the idea is to have one principal per OS user and we cannot have more than one JAAS file per JVM, so for cases where there is a producer and consumer in the same machine, it's just easier to have a single principal/JAAS file.

**Comment by ijuma** [comment]
@junrao Can you please merge this if you're happy with the latest changes (based on feedback from Magnus)?

**Comment by gwenshap** [comment]
LGTM

**Pull request closed without merging**